### PR TITLE
Dash implementation for uncertain delivery recovery.

### PR DESCRIPTION
**Pull request by ps2**
https://tidepool.atlassian.net/browse/LOOP-1750

### CODE REVIEW

File: DashKit/DashPumpManager.swift [hunk 1]
```diff
@@ -940,17 +980,170 @@ public class DashPumpManager: PumpManager {
...
+                    state.unfinalizedTempBasal?.cancel(at: commandDate)
+                }
+            }
+            state.pendingCommand = nil
+        }
+    }
+
+    public func attemptUnacknowledgedCommandRecovery() {
+        if let pendingCommand = self.state.pendingCommand {
+            podCommManager.queryAndClearUnacknowledgedCommand { (result) in
```
Comment (nhamming) [inline DashKit/DashPumpManager.swift:1115]: Wondering if there is any concern that the pending command in `state` doesn't match the unacknowledged command on the pump? Or am I overlooking some check that these are the same command?

File: DashKitUI/Views/DeliveryUncertaintyRecoveryView.swift [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    
+    let model: DeliveryUncertaintyRecoveryViewModel
+
+    init(model: DeliveryUncertaintyRecoveryViewModel) {
+        self.model = model
+    }
+
+    var body: some View {
+        GuidePage(content: {
+            Text("\(self.model.appName) has been unable to communicate with the pod on your body since \(self.uncertaintyDateLocalizedString).\n\nWithout communication with the pod, the app cannot continue to send commands for insulin delivery or display accurate, recent information about your active insulin or the insulin being delivered by the Pod.\n\nMonitor your glucose closely for the next 6 or more hours, as there may or may not be insulin actively working in your body that \(self.model.appName) cannot display.")
```
Comment (nhamming) [inline DashKitUI/Views/DeliveryUncertaintyRecoveryView.swift]: [nit] localization?

File: DashKitUI/Views/DeliveryUncertaintyRecoveryView.swift [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                .padding([.top, .bottom])
+        }) {
+            VStack {
+                Text("Attemping to restablish communication").padding(.top)
```
Comment (nhamming) [inline DashKitUI/Views/DeliveryUncertaintyRecoveryView.swift]: [nit] localization?

File: DashKitUI/Views/DeliveryUncertaintyRecoveryView.swift [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+                    self.model.podDeactivationChosen?()
+                }) {
+                    Text(LocalizedString("Deactivate Pod", comment: "Button title to deactive pod on uncertain program"))
+                    .actionButtonStyle()
+                    .padding()
+                }
+            }
+        }
+        .environment(\.horizontalSizeClass, horizontalOverride)
+        .navigationBarTitle(Text("Unable to Reach Pod"), displayMode: .large)
```
Comment (nhamming) [inline DashKitUI/Views/DeliveryUncertaintyRecoveryView.swift]: [nit] localization?

File: DashKitUI/Views/UncertaintyRecoveredView.swift [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+import LoopKitUI
+
+struct UncertaintyRecoveredView: View, HorizontalSizeClassOverride {
+    var appName: String
+    
+    var didFinish: (() -> Void)?
+    
+    var body: some View {
+        GuidePage(content: {
+            Text("\(self.appName) has recovered communication with the pod on your body.\n\nInsulin delivery records have been updated and should match what has actually been delivered.\n\nYou may continue to use \(self.appName) normally now.")
```
Comment (nhamming) [inline DashKitUI/Views/UncertaintyRecoveredView.swift:19]: [nit] localization?

File: DashKitUI/Views/UncertaintyRecoveredView.swift [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+                    self.didFinish?()
+                }) {
+                    Text(LocalizedString("Continue", comment: "Button title to continue"))
+                    .actionButtonStyle()
+                    .padding()
+                }
+            }
+        }
+        .environment(\.horizontalSizeClass, horizontalOverride)
+        .navigationBarTitle(Text("Comms Recovered"), displayMode: .large)
```
Comment (nhamming) [inline DashKitUI/Views/UncertaintyRecoveredView.swift:33]: [nit] localization?

**Review by nhamming — APPROVED** [review]
General comment: I found it was difficult to follow the returned DashPumpManager errors since they jump between `PumpManagerResult` and `DashPumpManagerError`.

Didn't detect any issues. LGTM

File: DashKit/DashPumpManager.swift (hunk 1 above)
Comment (ps2) [reply DashKit/DashPumpManager.swift:1115]: The pump won't accept any new commands when there is an unacknowledged command.

**Pull request merged**